src/lavatory/utils/artifactory.py

```diff
@@ -112,51 +102,61 @@ def filter(self, repo, terms=[], depth=3, sort=None, offset=None, limit=None):
         the default n items will be enough.
         """
 
+        if terms is None:
+            terms = []
+
         terms.append({"repo": {"$eq": repo}})
         terms.append({"type": {"$eq": "folder"}})
         terms.append({"depth": {"$eq": depth}})
 
-        findexpr = json.dumps({"$and": terms})
-    
-        aql = "items.find({})".format(findexpr)
-    
+        find_expr = json.dumps({"$and": terms})
+
+        aql = "items.find({})".format(find_expr)
+
         if sort:
             aql += ".sort({})".format(json.dumps(sort))
-    
+
         if offset:
             aql += ".offset({})".format(offset)
-    
+
         if limit:
             aql += ".limit({})".format(limit)
-    
+
         LOG.debug("AQL: {}".format(aql))
-    
-        response = self.http.post("/search/aql", aql)
+
+        response = self.artifactory.find_by_aql(criteria=aql)
 
         return response["results"]
-    
-    def retain(self, repo, specproject, depth, terms=None, count=None, weeks=None):
+
+    def retain(self, repo, spec_project, depth, terms=None, count=None, weeks=None):
```

Comment:
repo can be removed from this function and use`self.repo_name`

Reply:
changed to repo_name from init.